src/main/java/com/redhat/rhjmc/containerjfr/net/web/handlers/HealthGetHandler.java

```diff
@@ -0,0 +1,111 @@
+ * without restriction, including without limitation the rights to copy, create
+ * derivative works of, display, perform, and distribute the Software and make,
+ * use, sell, offer for sale, import, export, have made, and have sold the
+ * Software and the Larger Work(s), and to sublicense the foregoing rights on
+ * either these or other terms.
+ *
+ * This license is subject to the following condition:
+ * The above copyright notice and either this complete permission notice or at
+ * a minimum a reference to the UPL must be included in all copies or
+ * substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ * #L%
+ */
+package com.redhat.rhjmc.containerjfr.net.web.handlers;
+
+import java.io.IOException;
+import java.util.Map;
+
+import javax.inject.Inject;
+import javax.inject.Provider;
+
+import org.apache.http.client.methods.CloseableHttpResponse;
+import org.apache.http.client.methods.HttpGet;
+import org.apache.http.impl.client.CloseableHttpClient;
+import com.google.gson.Gson;
+
+import com.redhat.rhjmc.containerjfr.core.sys.Environment;
+import com.redhat.rhjmc.containerjfr.net.web.HttpMimeType;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.ext.web.RoutingContext;
+
+class HealthGetHandler implements RequestHandler {
+
+    static final String GRAFANA_DATASOURCE_ENV = "GRAFANA_DATASOURCE_URL";
+
+    private final Provider<CloseableHttpClient> httpClientProvider;
+    private final Environment env;
+    private final Gson gson;
+
+    @Inject
+    HealthGetHandler(Provider<CloseableHttpClient> httpClientProvider, Environment env, Gson gson) {
+        this.httpClientProvider = httpClientProvider;
+        this.env = env;
+        this.gson = gson;
+    }
+
+    @Override
+    public String path() {
+        return "/health";
+    }
+
+    @Override
+    public HttpMethod httpMethod() {
+        return HttpMethod.GET;
+    }
+
+    // try-with-resources generates a "redundant" nullcheck in bytecode
+    @SuppressFBWarnings("RCN_REDUNDANT_NULLCHECK_WOULD_HAVE_BEEN_A_NPE")
+    @Override
+    public void handle(RoutingContext ctx) {
+        boolean datasourceAvailable = false;
+
+        if (this.env.hasEnv(GRAFANA_DATASOURCE_ENV)) {
+            try (CloseableHttpResponse response =
+                    this.httpClientProvider
+                            .get()
+                            .execute(new HttpGet(this.env.getEnv(GRAFANA_DATASOURCE_ENV))); ) {
```

Comment:
If the env var is not set, or is blank what happens? Is a request still fired? I think in that scenario it's worth just short-circuiting and skipping the network request.

Reply:
This is inside an if block for env.hasEnv so it should be skipped when env isn't set. I think it's fine to continue without checking the value; it will just error out in it's attempt in the client code